src/JsonLd/Serializer/ItemNormalizer.php

```diff
@@ -60,7 +60,7 @@ public function supportsNormalization($data, $format = null)
      */
     public function normalize($object, $format = null, array $context = [])
     {
-        $resourceClass = $this->resourceClassResolver->getResourceClass($object, $context['resource_class'] ?? null, true);
+        $resourceClass = $this->resourceClassResolver->getResourceClass($object, get_class($object), true);
```

Comment:
This is wrong, the `resourceClassResolver` already does this. If you want to fix this, you should try to fix it [in the ResourceClassResolver](https://github.com/api-platform/core/blob/374b8c8919deea0f7763fafcf445362567dd72ab/src/Api/ResourceClassResolver.php#L41) class directly.